### PR TITLE
Fix: Bookstack Update Function

### LINKED ISSUE #757: BookStack script leaves users in a questionable state in regards to updates
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

Currently [the script](https://github.com/community-scripts/ProxmoxVE/blob/e4354c1d53ca37eab38d898a33644241d072d8a8/install/bookstack-install.sh#L48-L50) is downloading BookStack from its source ZIP on GitHub, then unzipping that into a `/opt/bookstack` directory.

When it comes to updates, I'm not sure what's expected to occur, but the update script is getting the new version ZIP from GitHub, then extracting this before moving it into the /opt/bookstack directory:

https://github.com/community-scripts/ProxmoxVE/blob/e4354c1d53ca37eab38d898a33644241d072d8a8/ct/bookstack.sh#L69-L71

From my understanding, and later test of the script, this results in the new version files just being moved within the existing install (for example, at `/opt/bookstack/BookStack-24.10.3` if going between the latest releases) so does not actually result in an update.

Based upon how the `.env` file is handled, i assume that the intent is for the original install folder to be wiped out first then replaced, (which would change the behaviour of the `mv` to make sense) but this would be dangerous if intended because I can't see any handling of other potential user files (Uploaded images, customizations, attachments etc...).
Maybe the intent is some kind of merge/overwrite, but then I can't see how core app file/folder removals would be handled as expected.

Generally, the approach taken (manual handling of the source ZIP archives) is not an approach that's officially supported in BookStack, nor is common or expected, so attempts like this need careful consideration. The official guidance currently uses git to manage the core code-base.

For context, I'm the maintainer of BookStack and have encountered this while attempting to support a user to update.

### What settings are you currently utilizing?

- [ ] Default Settings
- [X] Advanced Settings

### Which Linux distribution are you employing?

Debian 12

### If relevant, including screenshots or a code block can be helpful in clarifying the issue.

_No response_

### Please provide detailed steps to reproduce the issue.

_No response_

### PR DESCRIPTION
**Pull request by michelroegl-brunner**
> **🛠️ Note:**  
> We are meticulous about merging code into the main branch, so please understand that pull requests not meeting the project's standards may be rejected. It's never personal!  
> 🎮 **Note for game-related scripts:** These have a lower likelihood of being merged.

---

## ✍️ Description

Fixes the Bookstack Update Function.
- 

- - -

- Related Issue: #757 
- Related PR: # (if applicable)  
- Related Discussion: [Link](https://github.com/community-scripts/ProxmoxVE/discussions)  

---

## 🛠️ Type of Change
Please check the relevant options:  
- [X] Bug fix (non-breaking change that resolves an issue)  
- [ ] New feature (non-breaking change that adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to change unexpectedly)  
- [ ] New script (a fully functional and thoroughly tested script or set of scripts)  

---

## ✅ Prerequisites
The following steps must be completed for the pull request to be considered:  
- [X] Self-review performed (I have reviewed my code to ensure it follows established patterns and conventions.)  
- [X] Testing performed (I have thoroughly tested my changes and verified expected functionality.)  
- [X] Documentation updated (I have updated any relevant documentation)

---

## 📋 Additional Information (optional)
Provide any extra context or screenshots about the feature or fix here.

### CODE REVIEW

**Review by MickLesk — DISMISSED** [review]

File: ct/bookstack.sh [hunk 1]
```diff
@@ -37,30 +37,38 @@ function update_script() {
...
-    msg_info "Updating ${APP} to ${RELEASE}"
-    cp /opt/bookstack/.env /opt/.env
-    wget -q "https://github.com/BookStackApp/BookStack/archive/refs/tags/v${RELEASE}.zip"
-    unzip -q v${RELEASE}.zip
-    mv BookStack-${RELEASE} /opt/bookstack
-    mv /opt/.env /opt/bookstack/.env
+    
+    msg_info "Updating ${APP} to v${RELEASE}"
+    mv /opt/bookstack /opt/bookstack-backup
+    wget -q --directory-prefix=/root/ "https://github.com/BookStackApp/BookStack/archive/refs/tags/v${RELEASE}.zip"
```
Comment (havardthom) [inline ct/bookstack.sh:43]:
```suggestion
    wget -qP /opt "https://github.com/BookStackApp/BookStack/archive/refs/tags/v${RELEASE}.zip"
```

File: ct/bookstack.sh [hunk 2]
```diff
@@ -37,30 +37,38 @@ function update_script() {
...
+    unzip -q ~/v${RELEASE}.zip
```
Comment (havardthom) [inline ct/bookstack.sh:44]:
```suggestion
    unzip -q /opt/v${RELEASE}.zip -d /opt
```

File: ct/bookstack.sh [hunk 3]
```diff
@@ -37,30 +37,38 @@ function update_script() {
...
+    mv ~/BookStack-${RELEASE} /opt/bookstack
```
Comment (havardthom) [inline ct/bookstack.sh:45]:
```suggestion
    mv /opt/BookStack-${RELEASE} /opt/bookstack
```

File: ct/bookstack.sh [hunk 4]
```diff
@@ -37,30 +37,38 @@ function update_script() {
...
+    
     msg_info "Starting Apache2"
     systemctl start apache2
     msg_ok "Started Apache2"
-
+    
     msg_info "Cleaning Up"
-    rm -rf v${RELEASE}.zip
+    rm -rf /opt/bookstack-backup
+    rm -rf ~/v${RELEASE}.zip
```
Comment (havardthom) [inline ct/bookstack.sh:67]:
```suggestion
    rm -rf /opt/v${RELEASE}.zip
```

**Review by havardthom — COMMENTED** [review]
Can not assume that update script is being run as root user

**Review by MickLesk — DISMISSED** [review]
LGTM, work on my container

File: ct/bookstack.sh [hunk 5]
```diff
@@ -37,30 +37,38 @@ function update_script() {
...
-    cp /opt/bookstack/.env /opt/.env
-    wget -q "https://github.com/BookStackApp/BookStack/archive/refs/tags/v${RELEASE}.zip"
-    unzip -q v${RELEASE}.zip
-    mv BookStack-${RELEASE} /opt/bookstack
-    mv /opt/.env /opt/bookstack/.env
+    
+    msg_info "Updating ${APP} to v${RELEASE}"
+    mv /opt/bookstack /opt/bookstack-backup
+    wget -q --directory-prefix=/opt "https://github.com/BookStackApp/BookStack/archive/refs/tags/v${RELEASE}.zip"
+    unzip -q opt/v${RELEASE}.zip
```
[the remark this answers is not in the document]
Comment (michelroegl-brunner) [reply ct/bookstack.sh]: You are indeed correct. I missed the leading /. It is working now as intended.

[the remark this answers is not in the document]
Comment (michelroegl-brunner) [reply ct/bookstack.sh]: I´ve added the change to the develop branch for easy testing.

**Review by havardthom — APPROVED** [review]

**Review by MickLesk — APPROVED** [review]

**Pull request merged**